File: HavenApp/HavenApp/Views/VaultTypes.swift
import SwiftUI

// MARK: - Vault (Notes tab) type definitions
// Extracted from ViewerView for the Vault/Notes tab split.

enum ViewMode {
    case notes
    case media
    case likes
    case zaps
}

enum ContentFilter {
    case all
    case mine
    case tagged
    case whitelist
}

enum LikesFilter {
    case onMyNotes
    case onTagged
    case onWhitelisted
    case myLikes
}

enum ZapsFilter {
    case onMyNotes
    case onTagged
    case onWhitelisted
    case myZaps
}

enum SearchScope: CaseIterable, Equatable {
    case notes
    case profiles
    case hashtags

    var label: String {
        switch self {
        case .notes: return "Notes"
        case .profiles: return "Profiles"
        case .hashtags: return "Hashtags"
        }
    }

    var icon: String {
        switch self {
        case .notes: return "doc.text"
        case .profiles: return "person.2"
        case .hashtags: return "number"
        }
    }
}

struct ParsedZapReceipt {
    let senderPubkey: String
    let targetNoteId: String?
    let amountSats: Int64
}

File: NostrVault/app/src/main/java/com/nostrvault/data/model/VaultTypes.kt
package com.nostrvault.data.model

/**
 * Type definitions for the Vault / Relay tab, mirroring iOS VaultTypes.swift.
 */

enum class VaultViewMode(val displayName: String) {
    NOTES("Notes"),
    LIKES("Likes"),
    ZAPS("Zaps"),
}

enum class VaultContentFilter(val displayName: String) {
    ALL("All"),
    MINE("My Notes"),
    TAGGED("Tagged"),
    WHITELIST("Whitelisted"),
}

enum class VaultLikesFilter(val displayName: String) {
    ON_MY_NOTES("My Notes"),
    ON_TAGGED("Tagged"),
    ON_WHITELISTED("Whitelisted"),
    MY_LIKES("My Likes"),
}

enum class VaultZapsFilter(val displayName: String) {
    ON_MY_NOTES("My Notes"),
    ON_TAGGED("Tagged"),
    ON_WHITELISTED("Whitelisted"),
    MY_ZAPS("My Zaps"),
}

data class ParsedZapReceipt(
    val senderPubkey: String,
    val targetNoteId: String?,
    val amountSats: Long,
)
